compact.go

```diff
@@ -145,6 +146,22 @@ func (c *compactor) Plan() ([][]string, error) {
 		return [][]string{res}
 	}
 
+	// Check if any "large" blocks need to be recompacted due to deletions.
+	var compactDirs [][]string
+	for _, dm := range dms {
+		if dm.meta.MaxTime-dm.meta.MinTime > int64(c.opts.maxBlockRange/2) {
+			if dm.meta.Stats.NumTombstones > 0 &&
+				(dm.meta.Stats.NumSeries/dm.meta.Stats.NumTombstones) <= 20 { // 5%
+				compactDirs = append(compactDirs, []string{dm.dir})
+			}
+			continue
+		}
+		break
+	}
+
+	if len(compactDirs) > 0 {
+		return compactDirs, nil
+	}
```

Comment:
Probably better to place this after the regular compaction planning since we might already apply tombstones as a side effect of proper compactions.